File: src/main/org/nlogo/prim/etc/_magicopen.java
package org.nlogo.prim.etc;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;

public final strictfp class _magicopen
    extends Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax
        (new int[]{Syntax.TYPE_STRING},
            "O---", true);
  }

  @Override
  public void perform(final Context context)
      throws LogoException {
    perform_1(context, argEvalString(context, 0));
  }

  public void perform_1(final Context context, String name) {
    workspace.magicOpen(name);
    context.ip = next;
  }
}

File: src/main/org/nlogo/prim/etc/_magicopen.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Command, Context }

class _magicopen extends Command {
  override def syntax =
    Syntax.commandSyntax(
      Array(Syntax.StringType),
      "O---", true)
  override def perform(context: Context) {
    workspace.magicOpen(argEvalString(context, 0))
    context.ip = next
  }
}
